### PR TITLE
Edit New-AzSnapshot so it checks for already existing duplicate resource before creation

### PR DESCRIPTION
**Pull request by Sandido**
Work for item [https://msazure.visualstudio.com/One/_workitems/edit/8465720](https://msazure.visualstudio.com/One/_workitems/edit/8465720). 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 1]
```diff
@@ -1336,4 +1336,34 @@ function Test-DiskConfigTierSectorSizeReadOnly
...
+
+<#
+.SYNOPSIS
+
+#>
+function Test-SnapshotDuplicateCreationFails
+{
+    # Setup 
+    $rgname = Get-ComputeTestResourceName;
+    $loc = "northcentralus";#$loc = Get-ComputeVMLocation;
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1349]: cleanup

File: src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs [hunk 2]
```diff
@@ -41,6 +41,21 @@ public override void ExecuteCmdlet()
...
             ExecuteClientAction(() =>
             {
+                Snapshot existingResource;
+                try
+                { 
+                    existingResource = SnapshotsClient.Get(this.ResourceGroupName, this.SnapshotName);
+                }
+                catch
+                {
+                    existingResource = null;
```
Comment (Sandido) [inline src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs:51]: remove to its declaration

Comment (Sandido) [reply src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs:51]: no

File: src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs [hunk 3]
```diff
@@ -41,6 +41,21 @@ public override void ExecuteCmdlet()
...
+                }
+
+                if (existingResource != null)
+                {
+                    throw new Exception(string.Format("A Snapshot with name '{0}' in resource group '{1}' already exists. Please use Update-AzSnapshot to update an existing Snapshot.", this.SnapshotName, this.ResourceGroupName));
+                }
+
                 if (ShouldProcess(this.SnapshotName, VerbsCommon.New))
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs:59]: looks good. this should do the job. But I wonder how other resources check for existing resource

**Review by msJinLei — CHANGES_REQUESTED** [review]
@Sandido 
- Please fix the conflict of Changelog.md
- Please fix the following failed tests
```

  [xUnit.net 00:00:38.65]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineTests.TestVirtualMachineManagedDisk [FAIL]
    X Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineTests.TestVirtualMachineManagedDisk [1s 214ms]
EXEC : error Message:  [D:\a\1\s\build.proj]
     System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: A Snapshot with name 'ossnapshot' in resource group 'crptestps345' already exists. Please use Update-AzSnapshot to update an existing Snapshot.
    Stack Trace:
       at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
     at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
     at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
     at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
     at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
     at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
     at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
     at System.Management.Automation.PowerShell.Invoke()
     at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in D:\a\1\s\tools\ScenarioTest.ResourceManager\EnvironmentSetupHelper.cs:line 573
     at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\1\s\tools\TestFx\TestManager.cs:line 227
     at Microsoft.Azure.Commands.Compute.Test.ScenarioTests.VirtualMachineTests.TestVirtualMachineManagedDisk() in D:\a\1\s\src\Compute\Compute.Test\ScenarioTests\VirtualMachineTests.cs:line 270
    Standard Output Messages:
   Test-VirtualMachineManagedDisk
   Caught Exception: System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: A Snapshot with name 'ossnapshot' in resource group 'crptestps345' already exists. Please use Update-AzSnapshot to update an existing Snapshot.
      at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
      at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
      at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
      at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
      at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
      at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
      at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
      at System.Management.Automation.PowerShell.Invoke()
      at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in D:\a\1\s\tools\ScenarioTest.ResourceManager\EnvironmentSetupHelper.cs:line 573
   Message: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: A Snapshot with name 'ossnapshot' in resource group 'crptestps345' already exists. Please use Update-AzSnapshot to update an existing Snapshot.
EXEC : PowerShell error Record: A Snapshot with name 'ossnapshot' in resource group 'crptestps345' already exists. Please use Update-AzSnapshot to update an existing Snapshot. [D:\a\1\s\build.proj]
  Exception:System.Exception: A Snapshot with name 'ossnapshot' in resource group 'crptestps345' already exists. Please use Update-AzSnapshot to update an existing Snapshot.
      at Microsoft.Azure.Commands.Compute.Automation.NewAzureRmSnapshot.<ExecuteCmdlet>b__0_0() in D:\a\1\s\src\Compute\Compute\Generated\Snapshot\SnapshotCreateOrUpdateMethod.cs:line 59
      at Microsoft.Azure.Commands.Compute.ComputeClientBaseCmdlet.ExecuteClientAction(Action action) in D:\a\1\s\src\Compute\Compute\Common\ComputeClientBaseCmdlet.cs:line 61
      at Microsoft.Azure.Commands.Compute.Automation.NewAzureRmSnapshot.ExecuteCmdlet() in D:\a\1\s\src\Compute\Compute\Generated\Snapshot\SnapshotCreateOrUpdateMethod.cs:line 42
      at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
  Details:
  Script Stack Trace: at Test-VirtualMachineManagedDisk, D:\a\1\s\src\Compute\Compute.Test\bin\Debug\netcoreapp2.1\ScenarioTests\VirtualMachineTests.ps1: line 3615
   at <ScriptBlock>, <No file>: line 1
  : Target:
[xUnit.net 00:01:43.23]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.GalleryTests.TestGalleryImageVersionDiskImage [FAIL]
[xUnit.net 00:01:49.58]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.DiskRPTests.TestSnapshot [FAIL]
[xUnit.net 00:01:52.42]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.DiskRPTests.TestSnapshotConfigDiskAccessNetworkPolicy [FAIL]
[xUnit.net 00:01:53.18]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.DiskRPTests.TestSnapshotEncrypt [FAIL]
```

File: src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs (hunk 3 above)
Comment (Sandido) [reply src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs:59]: I don't think this pattern is commonly used yet, despite it being a [BP](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/new-cmdlet-example.md#new--cmdlet-example).

**Comment by Sandido** [comment]
@msJinLei , @isra-fel , is this considered a breaking change? This causes an error to occur when a duplicate resource already exists as opposed to just quietly failing the update and selecting the existing resource.

**Comment by msJinLei** [comment]
> @msJinLei , @isra-fel , is this considered a breaking change? This causes an error to occur when a duplicate resource already exists as opposed to just quietly failing the update and selecting the existing resource.

I think it is not a breaking change.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**